### PR TITLE
[risk=low][no ticket] Notebooks API: add tests and refactor

### PR DESCRIPTION
**Pull request by jmthibault79**
Follow-up to #7166 and #7196
* add API tests to show new cases where notebook file extensions are required
* add some .rmd tests
* rename some API method params to be more explicit about requiring extensions
* remove a hard-coded with-jupyter extension which is no longer needed
* NotebookUtils.withNotebookPath()
* other small refactoring

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/api/NotebooksControllerTest.java [hunk 1]
```diff
@@ -83,24 +83,6 @@ DbUser user() {
     }
   }
 
-  class MockNotebook {
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/NotebooksControllerTest.java:86]: moved to its own class in `utils`

File: api/src/test/java/org/pmiops/workbench/notebooks/NotebooksServiceTest.java [hunk 2]
```diff
@@ -106,25 +107,6 @@ DbUser getDbUser() {
     }
   }
 
-  class MockNotebook {
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/notebooks/NotebooksServiceTest.java:109]: duplicates the class in utils

**Review by evrii — APPROVED** [review]
Looks good to me. I ran the tests locally, and they all passed. Seems like good refactors. The only thing that I am unsure about is the set of functions that take filenames with extensions. I am curious if this will end up being an assumption that might not necessarily be true.

File: api/src/main/java/org/pmiops/workbench/api/WorkspaceAdminController.java [hunk 3]
```diff
@@ -67,9 +67,10 @@ public ResponseEntity<WorkspaceAuditLogQueryResponse> getAuditLogEntries(
   @Override
   @AuthorityRequired({Authority.RESEARCHER_DATA_VIEW})
   public ResponseEntity<ReadOnlyNotebookResponse> adminReadOnlyNotebook(
-      String workspaceNamespace, String notebookName, AccessReason accessReason) {
+      String workspaceNamespace, String notebookNameWithFileExtension, AccessReason accessReason) {
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/api/WorkspaceAdminController.java:70]: The purpose of this renaming is not to **enforce it for callers**, but rather as a warning to callers that this enforcement exists, on a lower level.  Here, the enforcement is done at `NotebooksServiceImpl.adminGetReadOnlyHtml()`

This is not my ideal solution, but I think it's an incremental improvement on #7166

**Pull request merged**